toolbox/deps_update/main.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+)
+
+var (
+	repo       = flag.String("repo", "", "Update dependencies of only this repository")
+	githubClnt *githubClient
+)
+
+// Generates the url to the remote repository on github
+// embedded with proper username and token
+func remote(repo string) (string, error) {
+	acct := *owner // passed in as flag in githubClient
+	token, err := getToken()
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("https://%s:%s@github.com/%s/%s.git", acct, token, acct, repo), nil
+}
+
+// Update the commit SHA reference in a given line from dependency file
+// to the latest stable version
+// returns the updated line
+func replaceCommit(line string, dep dependency) (string, error) {
+	commitSHA, err := githubClnt.getHeadCommitSHA(dep.RepoName, dep.ProdBranch)
+	if err != nil {
+		return line, err
+	}
+	idx := strings.Index(line, "\"")
+	return line[:idx] + "\"" + commitSHA + "\",", nil
+}
+
+// Get the list of dependencies of a repo
+// Assumes repo has been cloned to local
```

Comment:
cloned to local? or cloned locally? does this program assume that we are in root directory of the repo?